File: 102-二叉树的层次遍历.cpp
class Solution {
public:
    void levelorder(vector<vector<int>>& myvector, TreeNode* root, int depth){
        if(root){
            if(depth >= myvector.size())
                myvector.push_back(vector<int> {});

            myvector[depth].push_back(root->val);
            levelorder(myvector, root->left, depth + 1);
            levelorder(myvector, root->right, depth + 1);
        }
    }

    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        levelorder(result, root, 0);
        return result;
    }
};

File: 102-二叉树的层次遍历.py
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        import queue

        result = []
        myqueue = queue.Queue()
        myqueue.put((root, 0))
         
        while not myqueue.empty():
            node, depth = myqueue.get()
            if node:
                if depth > len(result) - 1:
                    result.append([node.val])
                else:
                    result[depth].append(node.val)
                myqueue.put((node.left, depth + 1))
                myqueue.put((node.right, depth + 1))
        
        return result
